File: src/jastudio/jastudio/note/notefields/strip_html_on_read_fallback_string_field.py
from __future__ import annotations

from typing import TYPE_CHECKING

from autoslot import Slots  # pyright: ignore[reportMissingTypeStubs]
from jastudio.note.notefields.fallback_string_field import FallbackStringField
from sysutils import ex_str

if TYPE_CHECKING:
    from jastudio.note.notefields.mutable_string_field import MutableStringField

class StripHtmlOnReadFallbackStringField(Slots):
    def __init__(self, primary_field: MutableStringField, fallback_field: MutableStringField) -> None:
        self._field: FallbackStringField = FallbackStringField(primary_field, fallback_field)

    def get(self) -> str: return ex_str.strip_html_markup(self._field.get().replace("<wbr>", ex_str.invisible_space))

File: src/src_dotnet/JAStudio.Core/Note/NoteFields/StripHtmlOnReadFallbackStringField.cs
namespace JAStudio.Core.Note.NoteFields;

public class StripHtmlOnReadFallbackStringField
{
   readonly FallbackStringField _field;

   public StripHtmlOnReadFallbackStringField(IStringValue primaryField, IStringValue fallbackField) => _field = new FallbackStringField(primaryField, fallbackField);

   public string Get() => StringExtensions.StripHtmlMarkup(_field.Get().Replace("<wbr>", StringExtensions.InvisibleSpace));
}
